tests/PHPStan/Rules/Comparison/data/loose-comparison-against-enums.php

```diff
@@ -106,4 +106,72 @@ public function nullableEnum(?FooUnitEnum $u, string $s): void
 		if (in_array($s, [$u])) {
 		}
 	}
+
+	/**
+	 * @param array<string> $strings
+	 * @param array<FooUnitEnum> $unitEnums
+	 */
+	public function dynamicValues(FooUnitEnum $u, string $s, array $strings, array $unitEnums): void
+	{
+		if (in_array($u, $unitEnums)) {
+		}
+
+		if (in_array($u, $unitEnums, false)) {
+		}
+
+		if (in_array($u, $unitEnums, true)) {
+		}
+
+		if (in_array($u, $strings)) {
+		}
+
+		if (in_array($u, $strings, false)) {
+		}
+
+		if (in_array($u, $strings, true)) {
+		}
+
+		if (in_array($s, $strings)) {
+		}
+
+		if (in_array($s, $strings, false)) {
+		}
+
+		if (in_array($s, $strings, true)) {
+		}
+
+		if (in_array($s, $unitEnums)) {
+		}
+
+		if (in_array($s, $unitEnums, false)) {
+		}
+
+		if (in_array($s, $unitEnums, true)) {
+		}
+	}
+
+    /**
+     * @param non-empty-array<FooUnitEnum::A> $nonEmptyA
+     * @return void
+     */
+    public function nonEmptyArray(array $nonEmptyA): void
+    {
+        if (in_array(FooUnitEnum::B, $nonEmptyA)) {
+        }
```

Comment:
This slightly weird order of the ifs is not accidental. I had to move this if first, otherwise phpstan would complain about it being always false with `treatPhpDocTypesAsCertain = false`.